tool/lib/commands/analyze.dart

```diff
@@ -41,27 +42,26 @@ class AnalyzeCommand extends Command {
 
       final progress = log.progress('  ${p.relativePath}');
 
-      final process = await Process.start(
-        sdk.dartToolPath,
-        ['analyze', '--fatal-infos'],
+      final process = await processManager.runProcess(
+        CliCommand.from(
+          sdk.dartToolPath,
+          ['analyze', '--fatal-infos'],
+        ),
         workingDirectory: p.packagePath,
       );
-      final Stream<List<int>> stdout = process.stdout;
-      final Stream<List<int>> stderr = process.stderr;
 
-      final int exitCode = await process.exitCode;
+      final stdout = process.stdout;
+      final stderr = process.stderr;
+      final int exitCode = process.exitCode;
 
       if (exitCode == 0) {
         progress.finish(showTiming: true);
       } else {
         failureCount++;
 
         // Display stderr when there's an error.
-        final List<List<int>> out = await stdout.toList();
-        final stdOutput = convertProcessOutputToString(out, '    ');
-
-        final List<List<int>> err = await stderr.toList();
-        final errorOutput = convertProcessOutputToString(err, '    ');
+        final stdOutput = convertProcessOutputToString(stdout, '    ');
+        final errorOutput = convertProcessOutputToString(stderr, '    ');
```

Comment:
do we even need these? I thought ProcessManager automatically shows stdout and stderr output

Reply:
Not as far as I can see. It used to run the process and return stdout, but if exitCode is non-zero through. It doesn't seem like it ever prints the output though (I have made some changes to it to return both stdout and stderr, but looking at the original code there was no printing their either).

Some of the commands tun through it (like fetching branches etc.) probably don't want to all be printed anyway?

Reply:
I've done this, though I think the original behaviour for some of these commands (like this) were to _not_ show stdout (and only show stderr if there was a failure) so the output might be more verbose now?

That might not be bad (it might even be good) on the bots, though might be less useful locally. I don't know if we can trivially suppress the stdout using `_ForwardingSpawn` though?

Reply:
For tools like this, I think more verbosity is better so we can quickly see where things go wrong 👍